West bash completion

carlescufi opened the pull request:
Fixes https://github.com/zephyrproject-rtos/west/issues/220

carlescufi committed:
west: Add bash completion file

This file was previously located in the west repository, under scripts/.
Since it now includes knowledge about specific behavior ef zephyr
extension commands, we move it here after overhauling it completely.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented on doc/guides/west/install.rst:
```diff
@@ -0,0 +1,86 @@
+West is written in Python 3 and distributed through `PyPI`_.
+Use :file:`pip3` to install or update west:
+
+On GNU/Linux::
```
Sure :)

carlescufi commented on doc/guides/west/install.rst:
```diff
@@ -0,0 +1,86 @@
+
+.. _west-shell-completion:
+
+Enabling shell completion
```
I disagree here. I think this belongs in the installation guide because it's the sort of thing that users associate with installing. Also, I want to add an additional comment about remembering to update the completion file every time the user updates west with `pip`, and it would be nice to keep those 2 concepts together.

mbolivar replied:
Then I think you can remove it from the miscellaneous file.

It's not the case that that's the catch-all location for documenting non build/flash/debug commands. The sign command has its own page, for example.

carlescufi replied:
Done

carlescufi commented on doc/guides/west/install.rst:
```diff
@@ -0,0 +1,86 @@
+    `~/.bash_profile`
+  * Install the ``bash-completion`` package with ``brew``::
+      brew install bash-completion``
+    then source the main bash completion script in your :file:`~/.bash_profile`::
```
I have some doc build issues that I didn't detect locally for some reason, I will check on those

carlescufi commented on scripts/west_commands/completion.py:
```diff
@@ -0,0 +1,51 @@
+
+        # Remember to update completion/west-completion.bash if you add or remove
+        # flags
+        parser.add_argument('shell', nargs=1, choices=['bash'],
```
It's not redundant, I get `b` instead of `bash` without it in `args.shell`

mbolivar replied:
Yep, a positional takes one argument

carlescufi replied:
Don't know, is it the default? Did not see that at first glance in the doc

mbolivar commented on scripts/west_commands/completion.py:
```diff
@@ -0,0 +1,51 @@
+            # Keep this in sync with the string in west-commands.yml.
+            'display shell completion scripts',
+            'Display shell completion scripts.',
+            accepts_unknown_args=False)
```
OK

carlescufi replied:
We have it in multiple other commands, perhaps a round of separate cleanup is in order.

mbolivar commented on doc/guides/west/install.rst:
```diff
@@ -0,0 +1,86 @@
+###############
+
+West is written in Python 3 and distributed through `PyPI`_.
+Use :file:`pip3` to install or update west:
```
OK

carlescufi replied:
Yes to `upgrade` here but `--upgrade` seems a bit too long for the common user

carlescufi commented:
@mbolivar tried to address your review comments.

carlescufi committed:
west: Add completion command

Add a completion command that dumps the contents of a shell
completion file present in the zephyr repository.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
doc: west: Document west installation in-depth

Create a new dedicated section for west installation that details
some of the finer aspects of the process and steps involved.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

dbkinder reviewed (approved):
doc changes LGTM, thanks.

carlescufi commented on scripts/west_commands/completion.py:
```diff
@@ -0,0 +1,51 @@
+from west import log
+from west.commands import WestCommand
+
+COMPLETION_PATH = 'scripts/west_commands/completion/west-completion'
```
Good point, will fix.

carlescufi commented on scripts/west_commands/completion.py:
```diff
@@ -0,0 +1,51 @@
+        cf = os.path.join(zb, *COMPLETION_PATH.split('/'))
+        cf += '.' + args.shell[0]
+        if not os.path.exists(cf):
+            log.die('Unable to find completion file: {}'.format(cf))
```
Sure, no objections, will fix.

carlescufi commented on doc/guides/west/install.rst:
```diff
@@ -0,0 +1,86 @@
+The following are the typical locations where the west files will be
+installed if you followed the instructions above:
+
+* GNU/Linux:
```
+1

carlescufi commented on doc/guides/west/install.rst:
```diff
@@ -0,0 +1,86 @@
+   See :ref:`gs_python_deps` for additional clarfication on using the
+   ``--user`` switch.
+
+The following are the typical locations where the west files will be
```
Nice suggestion! will do

carlescufi commented on doc/guides/west/install.rst:
```diff
@@ -0,0 +1,86 @@
+
+  * GNU/Linux: bash
+  * macOS: bash
+  * Windows: not available
```
No, but WSL does. I might add a comment a WSL down the line.
